### PR TITLE
 tlf_handle_resolve: support for resolving implicit team IDs

### PR DESCRIPTION
**Pull request by strib**
Previously we were only resolving from a TLF name -> implicit team ID.  But KBFS also needs to do the reverse, and the service just added an RPC to make it possible, so now we can make use of it.

Issue: KBFS-2621

### CODE REVIEW

**Comment by strib** [comment]
(I chickened out on enabled mdv4 yet.  Needs more testing.  Going to make another PR for a better error in this version once mdv4 is turned on at the server.)

File: libkbfs/kbpki_client.go [hunk 1]
```diff
@@ -65,6 +65,25 @@ func (k *KBPKIClient) ResolveImplicitTeam(
...
+		return ImplicitTeamInfo{}, err
+	}
+
+	assertions, suffix, err := tlf.SplitExtension(name)
+	if err != nil {
+		return ImplicitTeamInfo{}, err
+	}
+
+	return k.serviceOwner.KeybaseService().ResolveIdentifyImplicitTeam(
+		ctx, assertions, suffix, tlfType, false, "")
```
Comment (jzila) [inline libkbfs/kbpki_client.go]: Could we get a unified call that does these two things in one RPC?

File: libkbfs/keybase_daemon_local.go [hunk 2]
```diff
@@ -239,12 +239,11 @@ func (k *KeybaseDaemonLocal) Resolve(ctx context.Context, assertion string) (
...
 
-	// TODO(KBFS-2621): Resolve shouldn't work for implicit teams, but
-	// until CORE-6623 is done, this is required.
-	iti, err := k.localImplicitTeams.getLocalImplicitTeam(id.AsTeamOrBust())
-	if err == nil {
-		// An implicit team exists, so use the display name.
-		return iti.Name, id, nil
+	_, ok := k.localImplicitTeams[id.AsTeamOrBust()]
+	if ok {
+		// An implicit team exists, so Resolve shouldn't work.
```
Comment (jzila) [inline libkbfs/keybase_daemon_local.go]: Meaning we should be using `ResolveImplicitTeamByID`?

File: libkbfs/kbpki_client.go (hunk 1 above)
Comment (strib) [reply libkbfs/kbpki_client.go]: No, this is the way CORE wanted to do it.  It's ok.

File: libkbfs/keybase_daemon_local.go (hunk 2 above)
Comment (strib) [reply libkbfs/keybase_daemon_local.go]: Yes, will clarify.

File: libkbfs/tlf_handle_resolve.go [hunk 3]
```diff
@@ -262,33 +262,33 @@ func makeTlfHandleHelper(
...
+		iteamInfo, err := ruid.resolver.ResolveImplicitTeamByID(
+			ctx, ruid.id.AsTeamOrBust(), ruid.tlfType)
 		if err == nil {
 			return nameIDPair{
-				name: name,
-				id:   ruid.id,
-			}, keybase1.SocialAssertion{}, id, err
+				name: iteamInfo.Name,
+				id:   iteamInfo.TID.AsUserOrTeam(),
+			}, keybase1.SocialAssertion{}, iteamInfo.TlfID, nil
```
Comment (songgao) [inline libkbfs/tlf_handle_resolve.go]: Do we want to verify `ruid.id == itemInfo.TID.AsUserOrTeam()` here?

**Pull request merged**